src/go/rpk/pkg/cli/topic/produce.go

```diff
@@ -107,6 +116,18 @@ func newProduceCommand(fs afero.Fs, p *config.Params) *cobra.Command {
 				out.Die("invalid empty format")
 			}
 
+			isSchemaRegistry := schemaID >= 0 || keySchemaID >= 0
+			// If the user is using schema registry, we want to use the json
+			// format: %v{json} and %k{json}.
+			if isSchemaRegistry {
+				if strings.Contains(inFormat, "%v{") || strings.Contains(inFormat, "%k{") {
+					// Do nothing. We don't want to replace what the user sets.
+				} else {
+					inFormat = strings.Replace(inFormat, "%k", "%k{json}", 1)
+					inFormat = strings.Replace(inFormat, "%v", "%v{json}", 1)
+				}
```

Comment:
I'm being a bit overly paranoid here, but there's a solution, so may as well be: this wont catch `%v %v{}`.

However a "simple" (regexp) solution is:
```go
inFormat = regexp.MustCompile("%(v|k)([^{]|$)").ReplaceAllString(s1, "%$1{json}$2")
```
which can be split into multiple lines as needed

Reply:
I went with a one-liner here, thanks!